src/main/java/com/google/devtools/build/lib/bazel/repository/starlark/StarlarkBaseExternalContext.java

```diff
@@ -1054,6 +1064,42 @@ public void createFile(
     }
   }
 
+  @StarlarkMethod(
+      name = "getenv",
+      doc =
+          // Should we disclose the (intended) side-effect of tracking env vars in the marker file?
```

Comment:
IMO yes. I would even go as far as declaring the other way to track env vars deprecated. Although we'd also need to add some comments to `StarlarkOS.environ`, explaining that just accessing the `rctx.os.environ` dict doesn't track anything.

Reply:
Sure, I can amend docs at both sites (here and `StarlarkOS.environ`), but I'd prefer to leave the deprecation decision for a separate change. :)

Ex:
- Should a change to an ephemeral variable like a proxy URL really invalidate a repository?
- What does Bazel's security model say about taking credentials from the environment and persisting them to disk (marker file) in the clear? (I realize `--client_env` is dumped to `java.log`, and then there's #20126 . But I don't want to make any problem worse.)

If going the deprecation route, then maybe it'd be worth adding an `ephemeral` (e.g.) parameter that toggles between (a) just reading a single key from `envVariables` or (b) (a) and also establishing a SkyKey dependency and recording the key-value in the marker file?